File: viaPorter-Yuheng/viaPorter/app/src/main/java/com/example/viapatron2/service/viaPatronWorkerService.java
package com.example.viapatron2.service;

import android.app.Service;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;
import com.example.viapatron2.app.managers.APIManager;
import com.example.viapatron2.app.managers.DataManager;
import io.reactivex.disposables.Disposable;

import java.util.List;

/**
 * Created by Lim Zhiming on 7/1/19.
 */

public class viaPatronWorkerService extends Service {

    private static final String TAG = "AutoWorkerService";

    // Binder and managers
    private final IBinder mBinder = new LocalBinder();

    // todo: create classes for below managers
    private DataManager dataManager;
    private APIManager apiManager;
    //private SocketManager socketManager;

    // disposables
    private List<Disposable> disposables;

    // getters
    public DataManager getDataManager() {
        return dataManager;
    }
    public APIManager getApiManager() {
        return apiManager;
    }
//    public SocketManager getSocketManager() {
//        return socketManager;
//    }

    // Binder class
    public class LocalBinder extends Binder {
        public viaPatronWorkerService getService() {
            return viaPatronWorkerService.this;
        }
    }

    @Override
    public IBinder onBind(Intent intent) {
        return mBinder;
    }
}

File: viaPorter-Yuheng/app/src/main/java/com/example/viaporter/service/viaPatronWorkerService.kt
package com.example.viaporter.service

import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import com.example.viaporter.app.managers.APIManager
import com.example.viaporter.app.managers.DataManager
import io.reactivex.disposables.Disposable

/**
 * Created by Lim Zhiming on 7/1/19.
 */

class viaPatronWorkerService : Service() {

    // Binder and managers
    private val mBinder = LocalBinder()

    // todo: create classes for below managers
    // getters
    val dataManager: DataManager? = null
    val apiManager: APIManager? = null
    //private SocketManager socketManager;

    // disposables
    private val disposables: List<Disposable>? = null
    //    public SocketManager getSocketManager() {
    //        return socketManager;
    //    }

    // Binder class
    inner class LocalBinder : Binder() {
        val service: viaPatronWorkerService
            get() = this@viaPatronWorkerService
    }

    override fun onBind(intent: Intent): IBinder? {
        return mBinder
    }

    companion object {

        private val TAG = "AutoWorkerService"
    }
}
